kinto/core/views/batch.py

```diff
@@ -144,6 +144,11 @@ def post_batch(request):
             resp, subrequest = request.follow_subrequest(subrequest,
                                                          use_tweens=False)
         except httpexceptions.HTTPException as e:
+            # Since some request in the batch failed, we need to stop the parent request
+            # through Pyramid's transaction manager.
+            if e.status_code == 409 or e.status_code >= 500:
```

Comment:
Are you sure the 500 is necessary here? I'm really not sure, since the final batch response status will be 5XX, the classic pyramid_tm commit veto is going to play its role.
(also don't hesitate to add this as a comment..)